docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement clear list feature in the <code>Duke</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Duke</code> class
+  
+  - Cons: Handling the command under the <code>Duke</code> class results in longer methods. Thus, the code becomes 
+  harder to navigate and understand. 
+  
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+<!-- @@author kokjoon97 -->
+### 3.8 Set budget feature
+#### 3.8.1 Current implementation
+
+The set budget feature is implemented using a <code>SetBudgetCommand</code> class which extends the main
+<code>Command</code> class with a variable representing the budget amount.
+
+The <code>Shoco</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>SetBudgetCommand</code> object based on that user input.
+
+The <code>Shoco</code> class then calls the <code>execute</code> method of the <code>SetBudgetCommand</code> object
+which makes another call to the <code>setBudget</code> function of the <code>Budget</code> object 
+with the amount specified by the user for the budget.
+
+The following sequence diagram below shows how the set budget feature works. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/setdiagram.png)
+
+
+#### 3.8.2 Design considerations
+
+##### Aspect: Data structure to support the set budget feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>SetBudgetCommand</code>
+ 
+  - Pros: Easy to add the set budget feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement set budget feature in the <code>Shoco</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Shoco</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Shoco</code> class
+  simply executes those commands as black boxes, without worrying about their internal details
+<!-- @@author -->
+ 
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+### 3.9 Reset budget feature
+#### 3.9.1 Current implementation
+
+The reset budget feature is implemented using a <code>ResetBudgetCommand</code> class which extends the main
+<code>Command</code> class with a variable representing the budget amount.
+
+The <code>Duke</code> class first receives user input from the <code>Ui</code> class before it creates a 
```

Comment:
For section 3 in general (implementation), the explanations for each feature are good. I noticed that for the add feature, the implementation is explained in a list, but it is just paragraphs for some other features. I think that maybe doing all of the explanations like the add feature would be great, and then it would help with consistency. 

![image](https://user-images.githubusercontent.com/40775612/77995943-8bfc5780-72e1-11ea-98c7-8a0eff0f91eb.png)

![image](https://user-images.githubusercontent.com/40775612/77995952-8ef74800-72e1-11ea-8ab3-28734384aa12.png)

Reply:
Oh sorry, there are two screenshots, one for AddCommand and another for DisplayCommand. White background makes it look like there is only one 😅